src/cmd/main.go

```diff
@@ -278,3 +289,50 @@ func setDiscardVarPredicate() error {
 
 	return nil
 }
+
+func initStubs() error {
+	if linter.StubsDir != "" {
+		linter.InitStubs()
+		return nil
+	}
+
+	// Try to use embedded stubs (from stubs/phpstorm_stubs.go).
+	if err := loadEmbeddedStubs(); err != nil {
+		return fmt.Errorf("failed to load embedded stubs: %v", err)
+	}
+
+	return nil
+}
+
+func loadEmbeddedStubs() error {
+	filenames := stubs.AssetNames()
+	if len(filenames) == 0 {
+		return fmt.Errorf("empty file list")
+	}
+
+	var errorsCount int
+
+	readStubs := func(ch chan linter.FileInfo) {
+		for _, filename := range filenames {
+			data, err := stubs.Asset(filename)
+			if err != nil {
+				log.Printf("Failed to read embedded %q file: %v", filename, err)
+				errorsCount++
+				continue
+			}
+			ch <- linter.FileInfo{
+				Filename: filename,
+				Contents: data,
+			}
+		}
+	}
+
+	if errorsCount != 0 {
```

Comment:
There probably is a "race" here as the whole point why readStubs writes to a channel is that it is executed in a separate goroutine. You can try to run noverify with "-race" option to ensure that it is in fact a race condition here. You need to use atomic increment here probably.

Reply:
And I actually put check before parsing call.
Thanks for spotting that. I'll fix that (and race things as well).

Reply:
You need to read using atomic instructions as well :).